### PR TITLE
fs/smartfs : Fix last sector chain to wrong sector when modify contents

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
If user modify current contents to the end of contents, it will update
wrong value for last sector's nextsector

### CODE REVIEW

**Comment by amogh-samsung** [comment]
Good that this was found...
LGTM

File: os/fs/smartfs/smartfs_smart.c [hunk 1]
```diff
@@ -850,6 +862,7 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 				/* Now get the chained sector info and reset the offset */
 				sf->curroffset = sizeof(struct smartfs_chain_header_s);
 			}
+
```
Comment (amogh-samsung) [inline os/fs/smartfs/smartfs_smart.c:865]: Can this be removed?

Comment (Taejun-Kwon) [reply os/fs/smartfs/smartfs_smart.c:865]: yes. I will

**Comment by amogh-samsung** [comment]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**